src/shogun/distributions/classical/GaussianDistribution.cpp

```diff
@@ -142,7 +138,7 @@ SGVector<float64_t> GaussianDistribution::log_pdf_multiple(SGMatrix<float64_t> s
 
 	/* one quadratic part x^T C^-1 x for each sample x */
 	SGVector<float64_t> result(num_samples);
-	Map<VectorXd> eigen_result(result.vector, result.vlen);
+	SGVector<double>::EigenVectorXtMap eigen_result = result;
```

Comment:
@vigsterkr could we have a templated function that does this so it is a bit less redundant?
`auto eigen_result = result.eigen()` or `auto eigen_result = eigen(result)`

Reply:
@gf712 what do you think about that?

Reply:
@karlnapf I actually had a similar discussion with @vigsterkr. This would then potentially enable us to return the correctly aligned Eigen::Map, i.e. 16/32/64 byte boundary. But then again, with the graph stuff this might be not that urgent, as there all the alignment will happen behind the scences at runtime.
Anyway, should be possible to have a `constexpr EigenVectorXtMap SGVector::eigen()`?

Reply:
It is not urgent, but as people are doing this, we can at least make the code a tiny bit less verbose?
@Abdelrahman350 it would be cool if you added such a method to SGVector/SGMatrix

Reply:
Just to make it clear @karlnapf . What should the method do exactly?

Reply:
@karlnapf i have to check whether this could be somehow be made to auto-kick-in for functions where the arg is that... the nice thing is that this is super implicit :) i see the point of you wanting to have less verbose/less boilerplate code. but adding that function that you are mentioning actually adding redundancy to the SGVector codebase as you'll have 2 functions that are doing the very same thing :P

Reply:
@Abdelrahman350 it would save us from writing `SGVector<double>::EigenVectorXtMap` every time, the function would return the mapped type.
@vigsterkr yes automatic would be best of course :) But we actually don't have EigenMap arguments in any functions, usually these maps are created inside functions that take sgvector as arguments.... I mean it was just an idea. Some functions have like 10 such maps in them and writing `SGVector<double>::EigenVectorXtMap` 10 times takes quite some space. Maybe it would be a templated util function instead of an sgvector method....not sure

Reply:
Ok, @karlnapf I have an idea to implement a function `auto eigen = globalMapping(m)` and use function overloading to do the same issue with vectors also. 
`SGVector<double> globalMapping(SGVector<float64_t> result){ SGVector<double>::EigenVectorXtMap eigen_result = result 
}`

Reply:
give it a try and we can have a look :)